### PR TITLE
feat(Header): Add ability to switch saved/applied links on auth status

### PR DESCRIPTION
**Pull request by dborysiewicz**
Moved authStatusTypes out of /header to /private to make it reusable for both header & footer, added authStatus to Footer & FooterLink and lastly check authStatus to change saved & applied links depending on status in both Header & Footer.

Only issues I see in this one is that the moving of authStatusTypes might break other apps if people where reading them in?
The other is Footer now also requires authStatus to be passed in however it will fail gracefully and revert with the default props.

### CODE REVIEW

**Comment by markdalgleish** [comment]
@dborysiewicz just a heads up that I renamed this, since I'm pretty sure the forward slash in the scope trips up semantic release.

File: react/Footer/FooterLink/FooterLink.js [hunk 1]
```diff
@@ -14,6 +15,7 @@ export default function FooterLink({ secondary, partner, analytics, className, l
         linkRenderer({
           'data-analytics': analytics,
           className: styles.link,
+          href: authenticationStatus === AUTHENTICATED && authedHref || href,
```
Comment (markdalgleish) [inline react/Footer/FooterLink/FooterLink.js]: Could this be a ternary instead?

**Comment by mengtzu** [comment]
Agree with @markdalgleish  on the ternary, otherwise 👍

**Comment by markdalgleish** [comment]
@dborysiewicz what about recommended jobs?

**Comment by dborysiewicz** [comment]
Closing for now as the nudges is on hold/redesign :(

**Pull request closed without merging**